src/consumer/kafka.py

```
@@ -0,0 +1,328 @@
+        await self.consumer.start()
+
+        # register signals on the running loop
+        loop = asyncio.get_running_loop()
+        for sig in (signal.SIGTERM, signal.SIGINT):
+            try:
+                loop.add_signal_handler(sig, lambda s=sig: asyncio.create_task(self._handle_shutdown(s)))
+            except NotImplementedError:
+                # Fallback for Windows / non-main thread
+                main_loop = loop  # captured from outer scope
+                def _sync_shutdown_handler(_sig, _frame):
+                    main_loop.call_soon_threadsafe(
```

Ruff (0.8.2)

Function definition does not bind loop variable `main_loop`

(B023)